src/test/java/org/isf/admission/test/Tests.java

```diff
@@ -1287,6 +1290,48 @@ private Admission buildNewAdmission() throws Exception {
 				diseaseOut2, diseaseOut3, operation, dischargeType, pregTreatmentType,
 				deliveryType, deliveryResult, true);
 	}
+	
+	@Test
+	public void checkAdmisionWithDiseaseIpdIn() throws Exception  {
+		//Given
+		int id = setupTestAdmission(false, true);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		String diseaseCode = "20";
+		//when
+		Disease diseaseIn1 = diseaseManager.getDiseaseByCode(diseaseCode);
+		Disease diseaseIn11 = admission.getDiseaseIn();
+		admission.setDiseaseIn(diseaseIn1);
+		// Then
+		assertThatThrownBy(() -> admissionBrowserManager.newAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class);
+		admission.setDiseaseIn(diseaseIn11);
+	}
+	
+	@Test
+	public void checkAdmisionWithDiseaseIpdOut() throws Exception {
+		// Given
+		int id = setupTestAdmission(false, true);
+		Admission admission = admissionBrowserManager.getAdmission(id);
+		String diseaseCode = "20";
+		// When
+		Disease diseaseOut = diseaseManager.getDiseaseByCode(diseaseCode);
+		Disease diseaseOut1 = admission.getDiseaseOut1();
+		admission.setDiseaseOut1(diseaseOut);
+		Disease diseaseOut2 = admission.getDiseaseOut2();
+		admission.setDiseaseOut2(diseaseOut);
+		Disease diseaseOut3 = admission.getDiseaseOut3();
+		admission.setDiseaseOut3(diseaseOut);
+		// Then
+		assertThatThrownBy(() -> admissionBrowserManager.updateAdmission(admission))
+				.isInstanceOf(OHDataValidationException.class)
+				.has(
+						new Condition<Throwable>(
+								(e -> ((OHServiceException) e).getMessages().size() == 1), "Expecting single validation error")
+				);
+		admission.setDiseaseOut1(diseaseOut1);
+		admission.setDiseaseOut2(diseaseOut2);
+		admission.setDiseaseOut3(diseaseOut3);
+	}
```

Comment:
What is the purpose of these last three statements?

Reply:
this is because I've noticed that after capturing the assertion, the initial values are refixed. you can look at testMgrValidateMaternity(), for example.

Reply:
please what about this can I remove or live it?

Reply:
@dbmalkovsky sounds something we can tackle into OP-911 or different issue?